extensions/investspec/investspec.py

```diff
@@ -0,0 +1,422 @@
+    """Represent an arg type as a user-friendly string.
+
+    Args:
+        arg_type (str): the type to format
+
+    Returns:
+        formatted string that links to a description of the input type
+    """
+    # Represent the type as a string. Some need a more user-friendly name.
+    # we can only use standard docutils features here, so no :ref:
+    # this syntax works to link to a section in a different page, but it
+    # isn't universally supported and depends on knowing the built page name.
+    if arg_type == 'freestyle_string':
+        return f'`text <{INPUT_TYPES_HTML_FILE}#text>`__'
+    elif arg_type == 'option_string':
+        return f'`option <{INPUT_TYPES_HTML_FILE}#option>`__'
+    elif arg_type == 'boolean':
+        return f'`true/false <{INPUT_TYPES_HTML_FILE}#truefalse>`__'
+    elif arg_type == 'csv':
+        return f'`CSV <{INPUT_TYPES_HTML_FILE}#CSV>`__'
+    else:
+        return f'`{arg_type} <{INPUT_TYPES_HTML_FILE}#{arg_type}>`__'
+
+
+def format_units_string(unit):
+    """Represent a pint Unit as a user-friendly phrase.
+
+    Args:
+        unit (pint.Unit): the unit to format
+
+    Returns:
+        string describing the unit
+    """
+    # pluralize the first unit so that it reads more naturally
+    custom_unit_plurals = {
+        'foot': 'feet',
+        'degree_Celsius': 'degrees_Celsius'
+    }
+    units_string = str(unit)
+    if units_string == 'none':
+        return ''
+
+    words = units_string.split(' ')
+    first_unit = words[0]
+    # check if it has an irregular plural form
+    if first_unit in custom_unit_plurals:
+        words[0] = custom_unit_plurals[first_unit]
+    # for all others, add 's' to the end
+    else:
+        words[0] = words[0] + 's'
+    units_string = ' '.join(words)
+    # pint separates words with underscores
+    units_string = units_string.replace('_', ' ')
+    # represent exponents with a caret rather than asterisks
+    units_string = units_string.replace(' ** ', '^')
+    # remove spaces around slashes
+    units_string = units_string.replace(' / ', '/')
+    return units_string
+
+
+def format_required_string(required):
+    """Represent an arg's required status as a user-friendly string.
+
+    Args:
+        required (bool | str): required property of an arg. May be `True`,
+            `False`, or a conditional string.
+
+    Returns:
+        string
+    """
+    if required is True:
+        return 'required'
+    elif required is False:
+        return 'optional'
+    else:
+        # assume that the about text will describe the conditional
+        return 'conditionally required'
+
+
+def format_geometries_string(geometries):
+    """Represent a set of allowed vector geometries as user-friendly text.
+
+    Args:
+        geometries (set(str)): set of geometry names
+
+    Returns:
+        string
+    """
+    # sort the geometries so they always display in a consistent order
+    sorted_geoms = sorted(
+        list(geometries),
```

Comment:
The cast to `list` here isn't necessary, as passing a dict to `sorted` will always take the keys of the dict and iterate over them.  The list here is pretty small, so this is a very minor point.  But if `geometries` were a dict of thousands or millions of features, converting to a list could be quite memory-intensive.

Reply:
Good point, removed the `list`.